app.ts

```diff
@@ -0,0 +1,341 @@
+  resetaddress.innerHTML = address.value;
+
+  //modify the data
+  data[index[0]]["First Name"] = fname.value;
+  data[index[0]]["Last Name"] = lname.value;
+  data[index[0]]["Middle Name"] = mname.value;
+  data[index[0]]["Email"] = email.value;
+  data[index[0]]["Phone Number"] = number.value;
+  data[index[0]]["Address"] = address.value;
+
+  //finally getting the edit button back
+  document.getElementById(
+    `edit_wrapper_${index}`
+  ).innerHTML = `<button id="edit_${index}" class="btn btn-secondary" >edit</button>`;
+  document.getElementById(`edit_${index}`).addEventListener("click", editRow);
+  // }
+};
+
+var makeLayout = (items: object[]): string[] => {
+  let itemsToDisplay: string[] = items.map((item: object, index: number) => {
+    let user = new User(
+      item["First Name"],
+      item["Middle Name"],
+      item["Last Name"],
+      item["Email"],
+      item["Phone Number"],
+      item["Address"]
+    );
+    return `<tr key=${index} id="row_${index}">
+            <td id="fname_${index}">
+            <div id="value_fname_${index}">
+            ${user["First Name"]}
+            </div>
+            </td>
+            <td id="mname_${index}">
+            <div id="value_mname_${index}">
+            ${user["Middle Name"]}
+            </div>
+            </td>
+            <td id="lname_${index}">
+            <div id="value_lname_${index}">
+            ${user["Last Name"]}
+            </div>
+            </td>
+            <td id="email_${index}">
+            <div id="value_email_${index}">
+            ${user["Email"]}
+            </div>
+            </td>
+            <td id="number_${index}">
+            <div id="value_number_${index}">
+            ${user["Phone Number"]}
+            </div>
+            </td>
+            <td id="address_${index}">
+            <div id="value_address_${index}">
+            ${user["Address"]}
+            </div>
+            </td>
+            <td id="edit_wrapper_${index}" ><button id="edit_${index}" class="btn btn-secondary")">edit</button></td>
+            <td ><button id="delete_${index}" class="btn btn-danger">delete</button></td>
+            </tr>`;
+  });
+
+  let inputelement: string = `<tr>
+  <td><input type="text" id="input_fname_new" placeholder="First name" ></td>
+  <td><input type="text" id="input_mname_new" placeholder="Middle name" ></td>
+  <td><input type="text" id="input_lname_new" placeholder="Last name" ></td>
+  <td><input type="email" id="input_email_new" placeholder="Email" ></td>
+  <td><input type="number" id="input_number_new" placeholder="Phone number" ></td>
+  <td><input type="text" id="input_address_new" placeholder="Address" ></td>
+  <td colspan=2><button class="btn btn-primary btn-block" type="button" id="submitnewuser" >Add</button></td>
+  </tr>`;
+
+  itemsToDisplay.push(inputelement);
+
+  return itemsToDisplay;
+};
+
+//general way for selecting elements to iterate over
+export var callElements = (
```

Comment:
why export here ?

Reply:
I am using this in validate.ts

> where is webpack config ?



> where is webpack config ?

Resolved. That file was skipped as it was a *.js file

Reply:
Actually I was planning to use this in validate module, but then i skipped that part.